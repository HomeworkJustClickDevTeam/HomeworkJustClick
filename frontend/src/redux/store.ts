import {combineReducers, configureStore, PreloadedState} from "@reduxjs/toolkit"
import userStateReducer from "./userStateSlice"
import roleReducer from "./roleSlice"
import groupReducer from "./groupSlice"
import isLoadingReducer from "./isLoadingSlice"
import homePageInReducer from "./homePageInSlice"



const rootReducer = combineReducers({
  userState: userStateReducer,
  role: roleReducer,
  group: groupReducer,
  isLoading: isLoadingReducer,
  homePageIn: homePageInReducer
})
export const setupStore = (preloadedState?: PreloadedState<RootState>) =>{
  return configureStore({
    reducer: rootReducer,
    preloadedState
  })
}


export type RootState = ReturnType<typeof rootReducer>
export type AppStore = ReturnType<typeof setupStore>
export type AppDispatch = AppStore['dispatch']