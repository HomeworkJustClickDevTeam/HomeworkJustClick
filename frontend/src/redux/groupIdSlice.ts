import {createSlice, PayloadAction} from "@reduxjs/toolkit";
import {RootState} from "./store";

type GroupIdSliceType = number | null
const initialState = null as GroupIdSliceType

export const groupIdSlice = createSlice({
    name: 'groupId',
    initialState,
    reducers: {
        setGroupId: (state, action: PayloadAction<GroupIdSliceType>) => action.payload
    }
})

export const {setGroupId} = groupIdSlice.actions
export const selectGroupId = (state: RootState) => state.groupId
export default groupIdSlice.reducer