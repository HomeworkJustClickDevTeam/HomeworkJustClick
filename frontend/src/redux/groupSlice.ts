import { createSlice, PayloadAction } from "@reduxjs/toolkit"
import { GroupInterface } from "../types/GroupInterface"
import { RootState } from "./store"

type GroupSliceType = GroupInterface | null
const initialState = null as GroupSliceType
export const groupSlice = createSlice({
  name:'group',
  initialState,
  reducers:{
    setGroup:(state, action:PayloadAction<GroupSliceType>) => action.payload
  }
})

export const {setGroup} = groupSlice.actions
export const selectGroup = (state:RootState) => state.group
export default groupSlice.reducer