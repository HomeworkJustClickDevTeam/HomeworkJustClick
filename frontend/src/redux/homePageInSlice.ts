import { createSlice, PayloadAction } from "@reduxjs/toolkit"
import { RootState } from "./store"

const initialState: boolean = true
export const homePageInSlice = createSlice({
  name: 'homePageIn',
  initialState,
  reducers:{
    setHomePageIn:(state, action:PayloadAction<boolean>) => action.payload
  }
})

export const {setHomePageIn} = homePageInSlice.actions
export const selectHomePageIn = (state:RootState) => state.homePageIn
export default homePageInSlice.reducer