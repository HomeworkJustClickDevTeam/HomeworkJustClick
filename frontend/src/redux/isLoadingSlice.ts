import { createDraftSafeSelector, createSlice } from "@reduxjs/toolkit"
import { AppDispatch, RootState } from "./store"

const initialState: boolean = false

export const isLoadingSlice = createSlice({
  name: 'isLoading',
  initialState,
  reducers:{
    setIsLoading: (_, action) => action.payload
  }
})

export const {setIsLoading} = isLoadingSlice.actions

export const setIsLoadingInReducer = (newState:boolean) => {
  return (dispatch:AppDispatch) => {
    dispatch(setIsLoading(newState))
  }
}
export const selectIsLoading = (state:RootState):boolean => state.isLoading //Use when you want to obtain the state of isLoading
export default isLoadingSlice.reducer