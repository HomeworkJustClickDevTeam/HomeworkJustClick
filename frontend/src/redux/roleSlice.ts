import { createSlice, PayloadAction } from "@reduxjs/toolkit"
import { RootState } from "./store"


type RoleSliceType = string|null
const initialState = null as RoleSliceType

export const roleSlice = createSlice({
  name: 'role',
  initialState,
  reducers:{
    setRole:(state, action:PayloadAction<RoleSliceType>) => action.payload
  }
})

export const {setRole} = roleSlice.actions
export const selectRole = (state:RootState) => state.role
export default roleSlice.reducer