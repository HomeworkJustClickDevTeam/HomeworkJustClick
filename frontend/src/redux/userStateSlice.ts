import { createSlice, PayloadAction } from "@reduxjs/toolkit"
import { UserInterface } from "../types/UserInterface"
import { RootState } from "./store"

type UserStateSliceType = UserInterface | null
export const userStateSlice = createSlice({
  name: 'userState',
  initialState: null as UserStateSliceType,
  reducers: {
    logOut: () => null,
    setUser: (state, action: PayloadAction<UserStateSliceType>) => action.payload
  }
})

export const {logOut, setUser} = userStateSlice.actions
export const selectUserState = (state: RootState) => state.userState
export default userStateSlice.reducer