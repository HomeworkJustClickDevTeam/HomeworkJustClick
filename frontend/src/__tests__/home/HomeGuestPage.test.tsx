import HomeGuestPage from "../../components/home/HomeGuestPage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import { setupStore } from "../../redux/store";
import { BrowserRouter } from "react-router-dom";


it("checks if home guest page renders correctly", () => {
  const renderedHomeGuestPage = renderer
    .create(
      <Provider store={setupStore()}>
        <BrowserRouter>
          <HomeGuestPage></HomeGuestPage>
        </BrowserRouter>
      </Provider>
    )
  expect(renderedHomeGuestPage).toMatchSnapshot()
});
