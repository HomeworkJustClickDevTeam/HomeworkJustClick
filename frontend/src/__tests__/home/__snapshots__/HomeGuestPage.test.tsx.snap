// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if home guest page renders correctly 1`] = `
<div
  className="flex w-screen flex-col items-center text-center font-lato font-normal select-none"
>
  <img
    alt="Kółko po lewej stronie"
    className="fixed left-[12%]  bottom-[15%] scale-50 xl:scale-100 -z-50 "
    src={
      {
        "process": [Function],
      }
    }
  />
  <img
    alt="Kółko po prawej stronie"
    className="fixed right-0 top-[12%] scale-50 translate-x-[25%] xl:transform-none -z-50 "
    src={
      {
        "process": [Function],
      }
    }
  />
  <div
    className="pt-[20vh] pb-8 xl:pb-[8vh]"
  >
    <h1
      className="text-3xl xl:text-[81px] font-medium pb-[calc(4px+1vh)] tracking-wide "
    >
      HomeworkJustClick
    </h1>
    <p
      className="text-s xl:text-[36px] font-medium "
    >
      Prosto, szybko, wygodnie!
    </p>
  </div>
  <a
    href="/login"
    onClick={[Function]}
  >
    <button
      className=" border-main_blue w-40 h-8 rounded-[5px] border-2 xl:w-[288px] xl:h-[56px] xl:border-[3px] xl:rounded-[10px] text-main_blue xl:text-[28px] hover:bg-hover_gray hover:shadow-md active:bg-opacity-60"
      type="button"
    >
      Zaloguj się
    </button>
  </a>
  <p
    className="text-[8px] xl:text-base text-main_blue"
  >
    Nie pamiętasz hasła?
  </p>
  <p
    className="text-[8px] xl:text-base mt-4 md:mt-[5vh]"
  >
    Nie masz jeszcze konta?
  </p>
  <a
    href="/register"
    onClick={[Function]}
  >
    <button
      className=" bg-main_blue w-40 h-8 rounded-[5px]  xl:w-[288px] xl:h-[56px] text-white xl:text-[28px] hover:bg-hover_blue hover:shadow-md active:shadow-none"
      type="button"
    >
      Zarejestruj się
    </button>
  </a>
</div>
`;
